common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -59,18 +59,7 @@ class HTTPClient(
 
     @Throws(IOException::class)
     private fun readFully(inputStream: InputStream): String {
-        return readFully(buffer(inputStream))
-    }
-
-    @Throws(IOException::class)
-    private fun readFully(reader: BufferedReader): String {
-        val sb = StringBuilder()
-        var line = reader.readLine()
-        while (line != null) {
-            sb.append(line)
-            line = reader.readLine()
-        }
-        return sb.toString()
+        return buffer(inputStream).readText()
```

Comment:
This will read the whole text, it would not be that performant if we had multiple lines, but responses are usually minified anyway so it shouldn't really matter. This way we avoid removing the final new line from the body. CC @vegaro in case you have any thoughts about this approach.

Another thing I noticed is that we were not closing the `BufferedReader` but we are closing the `inputStream` later on, so I think we shouldn't be leaking it.

Reply:
I think this is much simpler 👍🏻